traefikee/templates/proxy/deployment.yaml

```diff
@@ -1,72 +1,79 @@
+{{- $proxies := list "proxy" }}
+{{- if .Values.proxyInternal }}
```

Comment:
That's far better but that's why I was talking about a hash (dict in helm).
In `values.yaml`:

```yaml
additionalProxies:
  internal:
    replicas: 3
    securityContext:
      runAsUser: 65532
```

and here:

```yaml
{{- $proxies := dict "proxy" .Values.proxy }}
{{- $proxies = merge $proxies .Values.additionalProxies }}
{{- range $proxyName, $proxy := $proxies }}
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: {{ $.Values.cluster  }}-{{ $proxyName }}
...
```

don't you think it's much more simpler ?

You can even think about removing all `*-internal.yaml` files, just range over all proxies definitions ...

With the use of merge function we can even think about merging default proxy configuration into the additional ones to allow users to only overwrite specific stuff.

Reply:
See my last commits